src/blr/cli/buildmolecules.py

```diff
@@ -225,32 +231,48 @@ def report_and_remove_all(self):
         self.cache_dict = dict()
 
 
-def write_molecule_stats(molecule_dict):
+def calculate_N50(lengths):
     """
-    Writes stats file for molecules and barcode with information like how many reads, barcodes, molecules etc they
-    have
+    Based on https://gist.github.com/dinovski/2bcdcc770d5388c6fcc8a656e5dbe53c
     """
+    lengths = sorted(lengths, reverse=True)
 
-    # Opening all files
-    molecules_per_bc = open("molecules_per_bc.tsv", "w")
-    molecule_stats = open("molecule_stats.tsv", "w")
+    csum = np.cumsum(lengths)
+    n2 = int(sum(lengths) / 2)
 
-    # Write headers
-    print(f"Barcode\tNrMolecules", file=molecules_per_bc)
-    print(f"Reads\tLength\tBarcode\tNrMolecules",
-          file=molecule_stats)
+    # get index for cumsum >= N/2
+    csumn2 = min(csum[csum >= n2])
+    ind = np.where(csum == csumn2)
+    return lengths[ind[0][0]]
```

Comment:
Possibly this could even be moved to `utils`, it's usually really nice to be able get an N50 for a lot of things.